src/sentry/types/region.py

```diff
@@ -167,10 +166,37 @@ def parse_raw_config(region_config: Any) -> Iterable[Region]:
             yield Region(**config_value)
 
 
-def load_from_config(region_config: Any) -> GlobalRegionDirectory:
+def _generate_monolith_region_if_needed(regions: Collection[Region]) -> Iterable[Region]:
+    """Check whether a default monolith region must be generated.
+
+    Check the provided set of regions to see whether a region with the configured
+    name is present. If so, return an empty iterator. Else, yield the newly generated
+    region.
+    """
+    if not settings.SENTRY_MONOLITH_REGION:
+        raise RegionConfigurationError(
+            "`SENTRY_MONOLITH_REGION` must provide a default region name"
+        )
+    if not regions:
+        yield Region(
+            name=settings.SENTRY_MONOLITH_REGION,
+            snowflake_id=0,
+            address=options.get("system.url-prefix"),
+            category=RegionCategory.MULTI_TENANT,
+        )
+    elif not any(r.name == settings.SENTRY_MONOLITH_REGION for r in regions):
+        raise RegionConfigurationError(
+            "The SENTRY_MONOLITH_REGION setting must point to a region name "
+            f"({settings.SENTRY_MONOLITH_REGION=!r}; "
+            f"region names = {[r.name for r in regions]!r})"
+        )
+
+
+def load_from_config(region_config: Any) -> RegionDirectory:
```

Comment:
Now that we know we aren't going to be reading region configuration from an environment variable/string should we could probably streamline config loading a bit too. We could do that separately too though.